gateway/src/connectors/uniswap/uniswap.ts

```diff
@@ -71,6 +82,77 @@ export class Uniswap implements Uniswapish {
     return this.tokenList[address];
   }
 
+  /**
+   * The user sets an array of direct pools in their config to be used to find
+   * the least expensive route for a trade. This creates the pairs to be used
+   * in the route calculation. We do this on initiation because it requires
+   * asynchronous network calls. This checks the pool actually exists on Uniswap.
+   * If it does not, then the pair is ignored.
+   */
+  public async initDirectPools() {
+    for (const pair of this._poolStrings) {
+      const splitPair = pair.split('-');
+      if (splitPair.length === 2) {
+        const base = splitPair[0];
+        const quote = splitPair[1];
+        const baseTokenInfo = this.ethereum.getTokenForSymbol(base);
+        const quoteTokenInfo = this.ethereum.getTokenForSymbol(quote);
+
+        if (baseTokenInfo !== null && quoteTokenInfo !== null) {
+          const baseToken = new Token(
+            this.chainId,
+            baseTokenInfo.address,
+            baseTokenInfo.decimals,
+            baseTokenInfo.symbol,
+            baseTokenInfo.name
+          );
+
+          const quoteToken = new Token(
+            this.chainId,
+            quoteTokenInfo.address,
+            quoteTokenInfo.decimals,
+            quoteTokenInfo.symbol,
+            quoteTokenInfo.name
+          );
+
+          const pool = await this.getPool(
+            baseToken,
+            quoteToken,
+            this._factoryAddress,
+            this._factoryAbi
+          );
```

Comment:
I believe this would as an extra delay.
Response from the Fetcher can also be used to infer if a pair exists.